### PR TITLE
加入分析器

### PR DESCRIPTION
**Pull request by walterlv**
image

image

image

image

### CODE REVIEW

File: src/dotnetCampus.CommandLine.Analyzer.VisualStudio/source.extension.vsixmanifest [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
+  <Metadata>
+    <Identity Id="dotnetCampus.CommandLine.Analyzer.5e5274c0-ef92-4d29-86d9-6b6a561bb7e1" Version="1.0" Language="en-US" Publisher="walterlv"/>
+    <DisplayName>Analyzer1</DisplayName>
+    <Description xml:space="preserve">This is a sample diagnostic extension for the .NET Compiler Platform ("Roslyn").</Description>
```
Comment (kkwpsv) [inline src/dotnetCampus.CommandLine.Analyzer.VisualStudio/source.extension.vsixmanifest:6]: Description？

File: src/dotnetCampus.CommandLine.Analyzer/Analyzers/OptionLongNameMustBePascalCaseCodeFixProvider.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                    {
+                        // PascalCase does not support digital as the first letter.
+                        isWordStart = true;
+                        continue;
+                    }
+                    else if (!char.IsUpper(c))
+                    {
+                        isFirstLetter = false;
+                        isWordStart = false;
+                        builder.Append(char.ToUpper(c, CultureInfo.InvariantCulture));
```
Comment (kkwpsv) [inline src/dotnetCampus.CommandLine.Analyzer/Analyzers/OptionLongNameMustBePascalCaseCodeFixProvider.cs:111]: ToUpperInvariant

File: tests/dotnetCampus.CommandLine.Analyzer.Test/dotnetCampusCommandLineAnalyzerUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
+﻿//using Microsoft.CodeAnalysis;
```
Comment (kkwpsv) [inline tests/dotnetCampus.CommandLine.Analyzer.Test/dotnetCampusCommandLineAnalyzerUnitTests.cs:1]: why comment？

Comment (walterlv) [reply tests/dotnetCampus.CommandLine.Analyzer.Test/dotnetCampusCommandLineAnalyzerUnitTests.cs:1]: The unit test packages that the default template references are all disappeared so no test codes can be compiled.

File: Directory.Build.props [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
   <PropertyGroup>
     <PackageOutputPath>$(MSBuildThisFileDirectory)bin\$(Configuration)</PackageOutputPath>
+    <LangVersion>latest</LangVersion>
+    <Authors>walterlv</Authors>
+    <Company>dotnet-campus</Company>
+    <PackageLicenseExpression>MIT</PackageLicenseExpression>
+    <PackageProjectUrl>https://github.com/dotnet-campus/dotnetCampus.CommandLine</PackageProjectUrl>
+    <RepositoryUrl>https://github.com/dotnet-campus/dotnetCampus.CommandLine.git</RepositoryUrl>
+    <RepositoryType>git</RepositoryType>
+    <Version>2.0.0</Version>
```
Comment (lindexi) [inline Directory.Build.props:11]: 推荐放在 Version 文件里面

**Pull request merged**